opm/common/data/SimulationDataContainer.cpp

```diff
@@ -75,6 +75,36 @@ namespace Opm {
     }
 
 
+    void SimulationDataContainer::setCellDataComponent( const std::string& key , size_t component , const std::vector<int>& cells , const std::vector<double>& values) {
+        auto& data = getCellData( key );
+        if (component >= m_num_phases)
```

Comment:
> I think the num_phases thing should be removed from this class.

That I agree; however - if we are to retain the concept of multi component fields - like the saturation, we must still do some sanity check of the input argument component.